p2p/syncer.go

```diff
@@ -560,6 +630,27 @@ func (s *Syncer) maintainHealthyPeerSet() {
 		} else {
 			// sleep on failure to avoid spinning unproductively
 			time.Sleep(time.Second)
+
+			// if we're already connected to the random peer from the store,
+			// try to request additional peers
```

Comment:
We should assume that the majority of `Connect` calls will fail, meaning we'll end up on this branch quite frequently. So calling `getPeers` each time is going to be overkill. We should probably keep track of which peers we've called `getPeers` on, and ensure that we only call it once per peer.